rta/src/main/java/com/gluonhq/richtextarea/model/UnitBuffer.java

```diff
@@ -116,6 +116,65 @@ public void append(List<Unit> units) {
         unitList.addAll(units);
     }
 
+    /**
+     * Inserts a Unit into the buffer after a given position within its
+     * internal length, splitting in two the unit found at that position,
+     * if needed
+     * @param unit the unit to insert into this buffer
+     * @param position the position within the buffer range
+     */
+    public void insert(Unit unit, int position) {
+        if (unit == null) {
+            return;
+        }
+        if (position < 0 || position > length()) {
+            return;
+        }
+        int accum = 0;
+        List<Unit> buffer = new ArrayList<>();
+        for (Unit u : unitList) {
+            buffer.add(u);
+            if (u.isEmpty()) continue;
```

Comment:
should an empty Unit be added into the buffer?

Reply:
We are basically replacing the old units in the unitBuffer with the new ones. If the buffer had an empty unit, it doesn't make a difference, it can be easily drop out.
Just recall that a `UnitBuffer` is an abstraction of a `String`, and in its simple form, if all its units are `TextUnit`, the combination of all those string _tokens_ is just a string.